Mixmatch/streetview_dataset/parse_TFRecord.ipynb

```diff
@@ -0,0 +1,221 @@
+    "\n",
+    "def parse_labels(image_features):\n",
+    "    img_name = str(image_features['image/source_id'].numpy()).split('/')[-1][:-1]\n",
+    "    confidence = tf.sparse_tensor_to_dense(image_features['image/object/bbox/confidence'], default_value=0).numpy()\n",
+    "    xmin = tf.sparse_tensor_to_dense(image_features['image/object/bbox/xmin'], default_value=0).numpy()\n",
+    "    ymin = tf.sparse_tensor_to_dense(image_features['image/object/bbox/ymin'], default_value=0).numpy()\n",
+    "    xmax = tf.sparse_tensor_to_dense(image_features['image/object/bbox/xmax'], default_value=0).numpy()\n",
+    "    ymax = tf.sparse_tensor_to_dense(image_features['image/object/bbox/ymax'], default_value=0).numpy()\n",
+    "    \n",
+    "    bbox = np.vstack((xmin, ymin, xmax, ymax)) # Left, Top, Right, Bottom\n",
+    "    \n",
+    "    return img_name, confidence, bbox\n",
+    "\n",
+    "def img2example(img, label):\n",
+    "    imgByteArr = io.BytesIO()\n",
+    "    img.save(imgByteArr, format='JPEG')\n",
+    "    imgByteArr = imgByteArr.getvalue()\n",
+    "\n",
+    "    example = tf.train.Example(features=tf.train.Features(feature={\n",
+    "        \"image\": tf.train.Feature(bytes_list=tf.train.BytesList(value=[imgByteArr])),\n",
+    "        \"label\": tf.train.Feature(int64_list=tf.train.Int64List(value=[label]))}))\n",
+    "    \n",
+    "    return example"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def write_TFRecord_UCF(parsed_image_dataset, folder_path, writer):\n",
+    "    for image_features in parsed_image_dataset:\n",
+    "        img_name, confidence, bbox = parse_labels(image_features)\n",
+    "\n",
+    "        if img_name:\n",
+    "            label = 0\n",
+    "            if confidence.size>0 and max(confidence)>0.9:\n",
+    "                label = 1\n",
+    "                pos = np.argmax(confidence)\n",
+    "                bbox = bbox[:, pos]\n",
+    "            elif confidence.size==0 or (confidence.size>0 and max(confidence)<0.1):\n",
+    "                label = 0\n",
+    "            else:\n",
+    "                continue\n",
+    "\n",
+    "            img_path = os.path.join(folder_path, img_name)\n",
+    "            try:\n",
+    "                img = Image.open(img_path, \"r\")\n",
+    "\n",
+    "                # Exclude all non RGB images\n",
+    "                if len(img.getbands()) != 3:\n",
+    "                    continue\n",
+    "\n",
+    "                if label:\n",
+    "                    img = img.crop(bbox)\n",
+    "                img = img.resize((64, 64))\n",
+    "\n",
+    "                example = img2example(img, label)\n",
+    "                writer.write(example.SerializeToString())\n",
+    "            except Exception as e:\n",
+    "                print(e)\n",
+    "                print(img_path + \" is not valid\")\n",
```

Comment:
Avoid using try-exception clause if possible: https://github.com/google/styleguide/blob/gh-pages/pyguide.md#24-exceptions. If you have to use it, minimize the try block.

In this case, if Image.open can return a success or failure status, use if condition instead. Otherwise, only put Image.open() in the try block.

Reply:
Only Image.open() is in the try-exception clause right now.